src/Utils/InterfaceValidator.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Utils;
+
+use WordPress\AiClient\Providers\Models\Contracts\ModelInterface;
+use WordPress\AiClient\Providers\Models\ImageGeneration\Contracts\ImageGenerationModelInterface;
+use WordPress\AiClient\Providers\Models\SpeechGeneration\Contracts\SpeechGenerationModelInterface;
+use WordPress\AiClient\Providers\Models\SpeechGeneration\Contracts\SpeechGenerationOperationModelInterface;
+use WordPress\AiClient\Providers\Models\TextGeneration\Contracts\TextGenerationModelInterface;
+use WordPress\AiClient\Providers\Models\TextToSpeechConversion\Contracts\TextToSpeechConversionModelInterface;
+use WordPress\AiClient\Providers\Models\TextToSpeechConversion\Contracts\TextToSpeechConversionOperationModelInterface;
+
+/**
+ * Utility class for validating model interface implementations.
+ *
+ * Centralizes interface validation logic to reduce code duplication
+ * and provide consistent error messages across the AI Client.
+ *
+ * @since n.e.x.t
+ */
+class InterfaceValidator
```

Comment:
A personal preference is for utility classes to have simpler names, often referring to what they're validating.

```suggestion
class Models
```

That way it reads nicely: `Models::validateTextGeneration($model)`

Curious if @felixarntz agrees. 😁

Reply:
@JasonTheAdams I like the idea I've implemented it here https://github.com/WordPress/php-ai-client/pull/55/commits/79f4eeb051e728e0f3230251820d2f1fcbc5c736

Reply:
I'm not sure about `Models` - that could mean anything related to models. It could be a container class for multiple models, for example.

For a class that provides static utility methods, if we want to keep them constrained to one purpose, the class name should clarify that purpose (e.g. `ModelValidator`), or if we want to have them have _any_ kind of utility methods related to the DTO, we should at least give them a suffix to indicate they're a utility class (e.g. `ModelUtil`).

WDYT?